.cbmc-batch/source/openssl/evp_override.c

```diff
@@ -187,18 +280,212 @@ struct evp_cipher_st {
  * These functions return an EVP_CIPHER structure that contains the implementation of the symmetric cipher.
  */
 const EVP_CIPHER *EVP_aes_128_gcm(void) {
-    static const EVP_CIPHER cipher = { EVP_AES_128_GCM };
+    static const EVP_CIPHER cipher = { EVP_AES_128_GCM, 128 };
     return &cipher;
 }
 const EVP_CIPHER *EVP_aes_192_gcm(void) {
-    static const EVP_CIPHER cipher = { EVP_AES_192_GCM };
+    static const EVP_CIPHER cipher = { EVP_AES_192_GCM, 128 };
     return &cipher;
 }
 const EVP_CIPHER *EVP_aes_256_gcm(void) {
-    static const EVP_CIPHER cipher = { EVP_AES_256_GCM };
+    static const EVP_CIPHER cipher = { EVP_AES_256_GCM, 128 };
     return &cipher;
 }
 
+/* Abstraction of the EVP_CIPHER_CTX struct */
+struct evp_cipher_ctx_st {
+    EVP_CIPHER *cipher;
+    int encrypt;
+    int iv_len;           // default: DEFAULT_IV_LEN.
+    bool iv_set;          // boolean marks if iv has been set. Default:false.
+    bool padding;         // boolean marks if padding is enabled. Default:true.
+    bool data_processed;  // boolean marks if has encrypt/decrypt final has been called. Default:false.
+    int data_remaining;   // how much is left to be encrypted/decrypted. Default: 0.
+};
+
+/*
+ * EVP_CIPHER_CTX_new() creates a cipher context.
+ */
+EVP_CIPHER_CTX *EVP_CIPHER_CTX_new() {
+    EVP_CIPHER_CTX *cipher_ctx = can_fail_malloc(sizeof(EVP_CIPHER_CTX));
+    if (cipher_ctx) {
+        cipher_ctx->iv_len         = DEFAULT_IV_LEN;
+        cipher_ctx->iv_set         = false;
+        cipher_ctx->padding        = true;
+        cipher_ctx->data_processed = false;
+        cipher_ctx->data_remaining = 0;
+        cipher_ctx->cipher         = NULL;
+    }
+    return cipher_ctx;
+}
+/*
+ * EVP_CipherInit_ex(), EVP_CipherUpdate() and EVP_CipherFinal_ex() are functions that can be used for
+ * decryption or encryption. The operation performed depends on the value of the enc parameter.
+ * It should be set to 1 for encryption, 0 for decryption and -1 to leave the value unchanged (the actual value of 'enc'
+ * being supplied in a previous call). Return 1 for success and 0 for failure.
+ */
+int EVP_CipherInit_ex(
+    EVP_CIPHER_CTX *ctx,
+    const EVP_CIPHER *cipher,
+    ENGINE *impl,
+    const unsigned char *key,
+    const unsigned char *iv,
+    int enc) {
+    assert(ctx != NULL);
+    assert(enc == 0 || enc == 1 || enc == -1);
+    if (enc != -1) {
+        ctx->encrypt = enc;
+    }
+    if (cipher) {
```

Comment:
What happens in the `enc == 1 || enc == 0` cases? Those don't seem to be covered?